### ISSUE TITLE
Crash with complex expressions inside BETWEEN

### ISSUE THREAD

**Issue by kdeun1** [sql-formatter-org/sql-formatter#523]
**Describe the bug**
A certain postgresql query statement causes a "script error" since version 10.0.0.
It doesn't happen in version 9.2.0.

**Expected behavior**
It should work fine on all versions.

**Actual behavior**
[NORMAL QUERY]
```
with A_TABLE as (
select
  1 as a
  ,2 as b
  ,now() as c
union all
select
  1 as a
  ,6 as b
  ,now() - interval'1 minute' as c
union all
select
  1 as a
  ,6 as b
  ,now() - interval'1 day' as c
)
select * from A_TABLE
where	a > 0
and		b > 0
and		c between (now() - interval'1 hour') and now()
```

[OTHER QUERY]
```
with A_TABLE as (
select
  1 as a
  ,2 as b
  ,now() as c
union all
select
  1 as a
  ,6 as b
  ,now() - interval'1 minute' as c
union all
select
  1 as a
  ,6 as b
  ,now() - interval'1 day' as c
)
select * from A_TABLE
where	a > 0
and		b > 0
and		c between now() - interval'1 hour' and now()
```

- test jsfiddle env : http://jsfiddle.net/kdeun1/n6bksrLw/6/

The above jsfiddle environment is version 10.0.0.
When testing with a 'NORMAL QUERY' and an 'OTHER QUERY', the results of the two queries are different.
**There is a difference of one parenthesis in the between clause!**

option object used in jsfiddle
```js
option : {
  language: 'postgresql',
  tabWidth: 2,
  keywordCase: 'upper',
  linesBetweenQueries: 2,
}
```

version | <= 9.2.0 | > 10.0.0 |
-------- | ---------- | ------------
NORMAL QUERY | O | O | 
OTHER QUERY | O | script error |

**Usage**
How are you calling / using the library? npm install
What SQL language(s) does this apply to? postgresql
Which SQL Formatter version are you using? 9.2.0 vs. 10.0.0

**Comment by nene** [comment]
After further investigation I see it's a slightly different bug from #500. It boils down to supporting more complex expressions inside BETWEEN, like:

```sql
SELECT 5 BETWEEN 1 + 2 AND 10 + 8
```

It's also simpler to fix.

**Comment by nene** [comment]
This should now be fixed in 12.0.2 release
